src/components/Calendar/index.jsx

```diff
@@ -0,0 +1,77 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+
+import NEXT_BUTTON from "@assets/nextButton.svg";
+import PREV_BUTTON from "@assets/prevButton.svg";
+import CalendarPage from "@components/Calendar/CalendarPage";
+import Icon from "@components/common/Icon";
+import { throttle } from "@utils/";
+import { getDirectionValue } from "@utils/calendar";
+
+import * as S from "./style";
+
+const Calendar = () => {
+  // pageIndex는 더 상위에서 관리해야 함. 초기화되면 안됨.
+  const [pageIndex, setPageIndex] = useState(0);
+  const [slideInfo, setSlideInfo] = useState({
+    translateX: -50 + pageIndex * 50,
+    direction: null,
+  });
+  const isMovePending = useRef(false);
+
+  // const handleMoveCalendar = useCallback(
+  //   throttle((movePoint, direction) => {
+  //     setSlideInfo(({ translateX }) => ({ translateX: translateX + movePoint, direction: direction }));
+  //     setPageIndex((prev) => prev + getDirectionValue(direction));
+  //   }, 300),
+  //   [slideInfo],
+  // );
+
+  const handleMoveCalendar = useCallback(
+    (movePoint, direction) => {
+      if (!isMovePending.current) {
+        isMovePending.current = true;
+        setSlideInfo(({ translateX }) => ({ translateX: translateX + movePoint, direction: direction }));
+        setPageIndex((prev) => prev + getDirectionValue(direction));
+      }
+    },
+    [isMovePending.current],
```

Comment:
```suggestion
    [],
```

ref는 `.current` 프로퍼티의 값이 변해도 재렌더링을 유발하지 않기에(리액트가 값이 변화했다는 걸 알지 못합니다) `useCallback` 의 의존성 배열에 명시해주지 않아도 됩니다. 

어떤 값이 변함에 따라 재렌더링을 발생시키고 싶다면, `useRef` 대신 `useState` 를 사용하시는 게 맞을 거 같아요. 이 경우엔 `useRef` 를 사용하는 게 맞을 거 같고, 의존성 배열에서 제거해주시는 편이 좋겠습니다.